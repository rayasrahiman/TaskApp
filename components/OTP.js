import React, {useState} from 'react';
import {View, Text, StyleSheet, Platform} from 'react-native';
import OtpInputs from 'react-native-otp-inputs';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function OTP({onCodeChanged, error, onFocus}) {
  const [isFocused, setIsFocused] = useState(false);
  return (
    <View style={styles.inputCont}>
      <View style={{marginVertical: hp(3)}}>
        <Text style={{fontSize: hp(3), color: Colors.black}}>
          {I18n.t('sixDigitCode')}
        </Text>
      </View>
      <OtpInputs
        handleChange={onCodeChanged}
        numberOfInputs={6}
        inputStyles={styles.otpTxt}
        inputContainerStyles={{
          ...styles.otpCont,
          ...{borderColor: error ? Colors.red : Colors.gray},
        }}
        onFocus={() => {
          onFocus();
          setIsFocused(true);
        }}
        onBlur={() => {
          setIsFocused(false);
        }}
      />
      {error && (
        <View style={styles.titleCont}>
          <Text style={styles.errTxt}>{error}</Text>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  inputCont: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  otpTxt: {
    textAlign: 'center',
    fontSize: hp(3),
  },
  otpCont: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: hp(5),
    borderWidth: hp(0.3),
    borderRadius: 5,
    width: wp(9),
    height: hp(9),
    marginHorizontal: wp(3.5),
  },
  titleCont: {
    marginHorizontal: Platform.OS === 'android' ? wp(5) : wp(3),
    marginBottom: hp(3),
  },
  errTxt: {
    color: Colors.red,
    fontSize: hp(3),
  },
});
