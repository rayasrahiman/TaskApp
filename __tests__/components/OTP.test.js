import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import OTP from '../../components/OTP';

describe('OTP', () => {
  test('should render OTP correctly', () => {
    const wrapper = render(<OTP />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
