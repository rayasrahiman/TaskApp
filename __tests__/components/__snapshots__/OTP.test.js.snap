// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTP should render OTP correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 40,
        }
      }
    >
      6-digit code
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 120,
            "marginBottom": 66.5,
            "marginHorizontal": 26.5,
            "width": 67.5,
          },
          false,
        ]
      }
    >
      <TextInput
        accessibilityLabel="otpInput-0"
        accessible={true}
        autoCapitalize="none"
        clearTextOnFocus={false}
        firstInput={true}
        keyboardType="phone-pad"
        maxLength={6}
        numberOfInputs={6}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        secureTextEntry={false}
        selectTextOnFocus={true}
        style={
          Object {
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        testID="otpInput-0"
        textContentType="none"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 120,
            "marginBottom": 66.5,
            "marginHorizontal": 26.5,
            "width": 67.5,
          },
          false,
        ]
      }
    >
      <TextInput
        accessibilityLabel="otpInput-1"
        accessible={true}
        autoCapitalize="none"
        autoFocus={false}
        clearTextOnFocus={false}
        firstInput={false}
        keyboardType="phone-pad"
        maxLength={1}
        numberOfInputs={6}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        secureTextEntry={false}
        selectTextOnFocus={true}
        style={
          Object {
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        testID="otpInput-1"
        textContentType="none"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 120,
            "marginBottom": 66.5,
            "marginHorizontal": 26.5,
            "width": 67.5,
          },
          false,
        ]
      }
    >
      <TextInput
        accessibilityLabel="otpInput-2"
        accessible={true}
        autoCapitalize="none"
        autoFocus={false}
        clearTextOnFocus={false}
        firstInput={false}
        keyboardType="phone-pad"
        maxLength={1}
        numberOfInputs={6}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        secureTextEntry={false}
        selectTextOnFocus={true}
        style={
          Object {
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        testID="otpInput-2"
        textContentType="none"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 120,
            "marginBottom": 66.5,
            "marginHorizontal": 26.5,
            "width": 67.5,
          },
          false,
        ]
      }
    >
      <TextInput
        accessibilityLabel="otpInput-3"
        accessible={true}
        autoCapitalize="none"
        autoFocus={false}
        clearTextOnFocus={false}
        firstInput={false}
        keyboardType="phone-pad"
        maxLength={1}
        numberOfInputs={6}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        secureTextEntry={false}
        selectTextOnFocus={true}
        style={
          Object {
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        testID="otpInput-3"
        textContentType="none"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 120,
            "marginBottom": 66.5,
            "marginHorizontal": 26.5,
            "width": 67.5,
          },
          false,
        ]
      }
    >
      <TextInput
        accessibilityLabel="otpInput-4"
        accessible={true}
        autoCapitalize="none"
        autoFocus={false}
        clearTextOnFocus={false}
        firstInput={false}
        keyboardType="phone-pad"
        maxLength={1}
        numberOfInputs={6}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        secureTextEntry={false}
        selectTextOnFocus={true}
        style={
          Object {
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        testID="otpInput-4"
        textContentType="none"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 120,
            "marginBottom": 66.5,
            "marginHorizontal": 26.5,
            "width": 67.5,
          },
          false,
        ]
      }
    >
      <TextInput
        accessibilityLabel="otpInput-5"
        accessible={true}
        autoCapitalize="none"
        autoFocus={false}
        clearTextOnFocus={false}
        firstInput={false}
        keyboardType="phone-pad"
        maxLength={1}
        numberOfInputs={6}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        secureTextEntry={false}
        selectTextOnFocus={true}
        style={
          Object {
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        testID="otpInput-5"
        textContentType="none"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
